src/interfaces/wallet.cpp

```diff
@@ -456,8 +491,32 @@ class WalletImpl : public Wallet
     CWallet& m_wallet;
 };
 
+class WalletClientImpl : public ChainClient
```

Comment:
IMO this could have a better name, currently this chain client "manages all wallets".